### PR TITLE
pkg/termios: add darwin support

### PR DESCRIPTION
**Pull request by rminnich**
Signed-off-by: Ronald G. Minnich <rminnich@gmail.com>

### CODE REVIEW

**Comment by rminnich** [comment]
The stty command builds and works with no changes now.

File: pkg/termios/sgtty_darwin.go [hunk 1]
```diff
@@ -0,0 +1,178 @@
+// Copyright 2015-2017 the u-root Authors. All rights reserved
```
Comment (rjoleary) [inline pkg/termios/sgtty_darwin.go:1]: 2021

File: pkg/termios/sgtty_darwin.go [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+	// evidently. WTF?
+	t.Ispeed = int(term.Ispeed)
+	t.Ospeed = int(term.Ospeed)
+	t.Row = int(w.Row)
+	t.Col = int(w.Col)
+
+	return &t, nil
+}
+
+// STTY uses a TTY * to set TTY settings on an fd.
```
Comment (rjoleary) [inline pkg/termios/sgtty_darwin.go:50]:
```suggestion
// STTY uses a *TTY to set TTY settings on an fd.
```

File: pkg/termios/termios_darwin.go [hunk 3]
```diff
@@ -0,0 +1,160 @@
+// Copyright 2015-2017 the u-root Authors. All rights reserved
```
Comment (rjoleary) [inline pkg/termios/termios_darwin.go:1]: 2021

File: pkg/termios/termios_darwin.go [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+// MakeSerialDefault updates the Termios to typical serial configuration:
+// - Ignore all flow control (modem, hardware, software...)
+// - Translate carriage return to newline on input
+// - Enable canonical mode: Input is available line by line, with line editing
+//   enabled (ERASE, KILL are supported)
+// - Local ECHO is added (and handled by line editing)
+// - Map newline to carriage return newline on output
+func MakeSerialDefault(term *Termios) *Termios {
+	t := *term
+	/* Clear all except baud, stop bit and parity settings */
```
Comment (rjoleary) [inline pkg/termios/termios_darwin.go:145]: Use // style comments

**Review by rjoleary — APPROVED** [review]

**Pull request merged**